retrymanager_default.go

```diff
@@ -37,7 +37,9 @@ func (rc *retryControllerDefault) isRetriableError(err error) bool {
 		errors.Is(err, memdx.ErrSyncWriteReCommitInProgress) ||
 		errors.Is(err, cbhttpx.ErrConnectError) ||
 		errors.Is(err, ErrVbucketMapOutdated) ||
-		errors.Is(err, ErrCollectionManifestOutdated)
+		errors.Is(err, ErrCollectionManifestOutdated) ||
+		errors.Is(err, ErrNoServerAssigned) ||
+		errors.Is(err, ErrRepeatedReplicaRead)
```

Comment:
I think this ErrRepeatedReplicaRead should probably be handled directly by the error handling in GetReplicas rather than being pushed up to the retry handler.  Why would we be retrying in this case?

Reply:
Because even when we have already read a replica, we want to keep trying in that replicaIds thread. Since if there is a rebalance then the same replicaID could be routed to a replica on a node that we have not yet read.